### PR TITLE
[build gbc] Add custom options for Stack

### PR DESCRIPTION
**Pull request by ara-ayvazyan**

### CODE REVIEW

File: compiler/stack_build.cmake [hunk 1]
```diff
@@ -24,7 +24,7 @@ endif()
 set (buildGhcOptions "-O2")
 
 execute_process (
-    COMMAND ${STACK_EXECUTABLE} build :${target} --no-run-tests ${stack_options} --ghc-options=${buildGhcOptions}
+    COMMAND ${STACK_EXECUTABLE} build :${target} --no-run-tests ${stack_options} --ghc-options=${buildGhcOptions} ${BOND_STACK_OPTIONS}
```
Comment (chwarr) [inline compiler/stack_build.cmake]: Neither `stack build` nor `stack path` take `--allow-different-user`, though neither seems to complain if `--allow-different-user` is passed to it.

File: compiler/stack_build.cmake [hunk 2]
```diff
@@ -11,7 +11,7 @@
 cmake_policy (SET CMP0012 NEW)
 
 execute_process (
-    COMMAND ${STACK_EXECUTABLE} setup
+    COMMAND ${STACK_EXECUTABLE} setup ${BOND_STACK_OPTIONS}
```
Comment (chwarr) [inline compiler/stack_build.cmake]: `stack` has two different kinds of options: those that go between `stack` and the command (like `build` or `setup`) and those that go with a corresponding command (like `--allow-different-user` that the `setup` command [among others?] takes). This appears to only be for command-specific options, but is used across many different commands.

Which stack commands need `--allow-different-user`?

Comment (ara-ayvazyan) [reply compiler/stack_build.cmake]: I think `--allow-different-user` is not specific to `setup`. I can see it in `stack --help`.

Comment (chwarr) [reply compiler/stack_build.cmake]: Yup. I misread the help. This should then be:

```
COMMAND ${STACK_EXECUTABLE} ${BOND_STACK_OPTIONS} setup
```

File: compiler/stack_build.cmake (hunk 1 above)
Comment (ara-ayvazyan) [reply compiler/stack_build.cmake]: Both need it, because that is not a command-specific flag.

File: compiler/stack_build.cmake (hunk 2 above)
Comment (ara-ayvazyan) [reply compiler/stack_build.cmake]: Updated.

**Comment by tstein** [comment]
Last travis build job timed out at the last second. Merging.

**Pull request merged**